libs/user_extensions/src/TTAlpsEvent.cpp

```diff
@@ -125,3 +124,90 @@ bool TTAlpsEvent::ParticleHasISRmotherAfterTopMother(int particleIndex) {
 
   return ParticleHasISRmotherAfterTopMother(motherIndex);
 }
+
+bool TTAlpsEvent::IsGoodMuonFromALP(int muonIndex) {
+
+  auto genParticles = event->GetCollection("GenPart");
+  auto muon = asNanoGenParticle(genParticles->at(muonIndex));
+  
+  if (!muon->IsLastCopy()) return false;
+  if(!muon->IsMuon()) return false;
+
+  int motherIndex = muon->GetMotherIndex();
+  if (motherIndex < 0) return false;
+  auto mother = asNanoGenParticle(genParticles->at(motherIndex));
+
+  // loop over pythia copies of the muon
+  if (mother->IsMuon() == 13) {
```

Comment:
Wait, what's going on here? Doesn't `IsMuon()` return a bool? I mean, it will still work, since "13" evaluates to "true" in this case, but it looks evil.